### ISSUE TITLE
How many people will we have at the workshop?

### ISSUE THREAD

**Issue by connolly** [LSSTScienceCollaborations/ObservingStrategy#6]
Looking through the number of names that we will be contacting, we have about 24 people and 6 WGs. How many people/WGs can we reasonably support (and ensure a good cross-section of opinions and views)? If the work is going to be quantitative and we are going to have people experiment as they go through the discussions (e.g. querying the opsimDB to look at the distribution of observations in the galactic plane) it will require one or two people skilled in opsim/maf per WG. There are about 8-10 people with these skills available which means we are close to our limit of WGs.

Purely in terms of support, one person in a group should be able to help 3-4 people (i.e. to show how to run a query or just run it for them). This would limits our WGs to 8 or less people or a total of 40 people at the workshop. Are people comfortable with that number?

**Comment by ivezic** [comment]
On Mar 2, 2015, at 2:01 AM, Andrew Connolly <notifications@github.com> wrote:
> Looking through the number of names that we will be contacting, we have about 24 people and 6 WGs. How many people/WGs can we reasonably support (and ensure a good cross-section of opinions and views)? If the work is going to be quantitative and we are going to have people experiment as they go through the discussions (e.g. querying the opsimDB to look at the distribution of observations in the galactic plane) it will require one or two people skilled in opsim/maf per WG. There are about 8-10 people with these skills available which means we are close to our limit of WGs.
> 

Let’s discuss this in more detail. First to clarify: those names are candidates for leads on 
each topic, who will take responsibility that their science case is adequately represented
in the white paper. We have many because it’s not clear yet what fraction will accept our
generous offer to do work for free. Hence, it’s not known yet what fraction of those 24 will 
actually make it to the workshop. 

I am hoping that the group talking about science drivers in open cadence areas will not 
need a lot of interaction with opsimDB. One of the reasons for having two parallel tracks 
at the workshop was to avoid getting lost in technical details with MAF, and to focus instead 
on science drivers (including revisions due to progress in each area over the last decade 
or so). We will have a plenary talk about new Tier 1 cadence list, see 
https://confluence.lsstcorp.org/display/SIM/Cadence+Workshop+Simulated+Surveys

In this talk, I plan to emphasize open areas and specific questions that the Project needs
answered. I think this should be sufficient to engender good discussion, but please 
chime in if you think otherwise. 

> Purely in terms of support, one person in a group should be able to help 3-4 people (i.e. to show how to run a query or just run it for them). This would limits our WGs to 8 or less people or a total of 40 people at the workshop. Are people comfortable with that number? 
> 
> 


For the other track that would focus on MAF, we do have to be careful about the number
of people. I was hoping that you and Debbie would propose an upper limit and an 
algorithm for selecting attendees. 

I would be very surprised if we get 40 people for the MAF track. Actually, now I am 
beginning to worry if we will get enough people to make it worthwhile.

SOC, please comment! 

          Cheers, Zeljko

**Comment by drphilmarshall** [comment]
I'd suggest that rather than thinking in terms of parallel tracks, we instead think of *pairing up* (or at least grouping) domain experts (who may or may not know their way around the MAF) with MAF coders (who do). This connection is so important, it's almost the entire reason for the workshop! The pre-workshop homework for the domain experts could be to write pseudocode for the metrics they want to see implemented, and they could do that  homework in collaboration with their coding partners. Maybe it's enough for the domain experts to see some notebook-style MAF demos in advance, so that they understand what MAF does even if they aren't able to write metric code themselves. 

To me, 8-10 groups of people with 4-5 people in each group seems sensible, provided 2-3 people in each group can write MAF code. That would make roughly 40 people, with 20 of each type. If there's more MAF hackers than that, it'd be because more of the domain experts had learned the MAF, and that'd be a bonus.

**Comment by connolly** [comment]
I dont think we should have two tracks - I didnt see a proposal for a two
track set up in the notes that were sent round.

If we have two tracks we would end up with everyone going to the "science"
one and no one working on metrics or anything quantitative. Instead it
would be better if  the focus of our discussions in this workshop was on
quantitative assessments. This would help build off some of the outcomes of
the last cadence workshop, which went through a lot of the discussion we
are talking about here. For example, it is not clear to me that we really
need to have more discussion on the need for dithering but it would be
great if we could plan out what dithering experiments we should do to
define an optimal dithering pattern for the project and which metrics will
care about this. This discussion could be guided by seeing what MAF can
deliver and experimenting with dithering patterns in the current opsim
runs. Similar discussions could be held for the rolling cadence. If we do
this we will help address the first objective of this meeting which is "to
involve more people in defining more MAF metrics". These are the metrics
that will help in the definition of the next scheduler algorithms over the
coming year and I think these metrics should progress at the same pace as
the whitepaper (so that the whitepaper will have its maximal impact on the
cadence/scheduler development).

cheers
Andy

**Comment by ivezic** [comment]
On Mar 2, 2015, at 9:50 AM, Andrew Connolly <notifications@github.com> wrote:
> I dont think we should have two tracks - I didnt see a proposal for a two
> track set up in the notes that were sent round.

The two tracks idea was proposed at the last PST meeting after a short discussion.
It was meant to mitigate the problem of having people with clear science ideas but
being unable or unwilling to code up their MAF implementation. 

Given what Phil and Andy said, I have no problem with backpedalling on this 
two-track idea and instead having well-mixed groups per Phil�s proposal. In other
words, it�s precisely up to this SOC to optimize this workshop. Great discussions
and progress so far! 

      Cheers, Zeljko

**Comment by StephenRidgway** [comment]
Two “tracks” may still be effective if viewed as sequential rather than parallel.  At the first workshop, the prefatory MAF session was intended to bring a lot of people up to speed to code metrics.  This did not work as effectively as hoped, but that doesn’t mean it was not valuable or that the idea was wrong.  The same objective could drive a pre-workshop MAF session in 2015.  I’m sure there are a range of opinions.  I would suggest that the MAF session might last ~1.5 days in order to allow the kind of independent work that some people need to master a new topic.

**Comment by StephenRidgway** [comment]
The intention of the 2014 workshop was to have a MAF-literate cadre ready to work with science teams and to code metrics during the workshop.  The return of metrics from that was meagre.  One option is to conclude that it failed, and to take another approach for 2015.  For example, the MAF training session could be separated out and run weeks or months in advance.  This could include coding of some of the metrics that have been identified by algorithm but not yet implemented, and thus make some progress straight-away.  Work in this venue could be the entree to travel support for MAF-gurus to attend the main workshop.

**Comment by drphilmarshall** [comment]
I like this idea very much. I thought the pre-workshop MAF "dry run" was
really helpful in getting started with the coding.

**Comment by connolly** [comment]
I agree, running a training session ahead of time would be good to ensure that we have enough people ready to go. Would we do this in person or remotely (I assume the latter). The only question would be how much burden it puts on the people who will have to run this session.

Perhaps we could go one step further and make it a requirement for anyone who wants to take part in the workshop - guru or not. It would demonstrate a commitment to doing work at the workshop

**Comment by djbard** [comment]
One of the problems with the 2014 workshop was that there wasn't enough time to actually *code* metrics - instead the discussion was mostly science-related, similar to the 'science track' focus discussed above. In the groups I was involved in, after we'd discussed/identified the science metrics there simply wasn't enough time to code them up. Once we'd left the meeting, of course we never got round to coding them either. I'd like to ensure we have enough time to code up metrics *during* the workshop, beyond just discussing them. 
FWIW, I also dislike the idea of two separate tracks, but I do like the concept of having a pre-workshop training session so we all arrive ready to work.

**Comment by bethwillman** [comment]
Hi all -

Let me simultaneously dip my toe into GitHub issues and throw a monkey wrench into things.  

I'm concerned about the model where the primary way to communicate quantitative science case needs is through direct coding of MAF metrics, which is how this conversation is reading to me and how I interpreted last year's approach.

**This approach seems inefficient:** If 40 people each spend 20 hours this one week to learn how to, and then to code up, metrics - that is nearly 0.5 FTE.  If we include people installing MAF on their machines, time spent by Project members debugging that, and time spent prior to the workshop doing tutorials... we're probably at nearly 1 FTE. Then add in the time spent at least year's workshop and the time to be spent in the coming years.  All of this time invested, and only a very small fraction of scientists who we want to engage in the process have actually been engaged.  

The Project could/should pay 2 people x 2 years x 0.5 FTE to simply code in metrics for people.  Then super-users who desire could take an "MAF track" and really get their hands dirty and contribute to that effort, and the rest of us could engage on the science and pseudocode level that we're used to. 

**This approach seems alienating:**  If we send the message that people need to install MAF, engage in tutorials, and/or spend a day long on technical training to attend this workshop -- Then who will come?  How will we retain people's interest in participating?  When I go to TMT/GMT etc workshops - there is money to pay for my trip, there is alcohol, there is free food, and I feel there is a zero level entry point to get my voice heard.  And I'm not even on the team.  We can't do most of that, so we have to do an amazing job at the latter :)

(Along these lines, I don't think we should add a II onto last years conference title because I think we want to attract an expanded and different type of audience.)

I read the "mixed tracks" suggestion to be effectively similar to me as single track.  I need to learn new technical stuff before getting to meat.  My initial support for Steve's suggestion of sequential tracks gave way to skepticism when I thought about who would then attend the workshop - We would first engage people who have time to learn a specialized technical skill and use it to plan for LSST in a way that is unlikely to lead to cited papers (all unpaid effort).  Who falls into that category?  I don't, and neither do many the people who I would call to ask for input on LSST cadence planning.  This isn't to say there aren't many amazing amazing people who do fall into that category (including people on this SOC), but I think that is a narrow set of the community.

I strongly advocate for two tracks at this workshop, with the MAF track being for people who desire to be super users (even if they aren't already, then they are welcome to learn) and the primary track being people who come ready to work/contribute *using tools they are already comfortable with* e.g. pencils, brains, loud mouths, latex typing, and pseudo coding.  

I'm quite interested to hear everyone's thoughts on this.  

Cheers
Beth

**Comment by StephenRidgway** [comment]
Beth,

Yes, somebody seemed to suggest that attendance should be limited to MAF coders, but I didn’t hear a “second” to that and you and I certainly don’t agree.

What I think some of the SOC were starting to agree on was having an early, separate session to bring up to speed a number of serious MAF coders.  I believe that numbers on the order of 10-20 seemed plausible.  That event would not be the workshop per se, but a preparatory session, which would help to ensure human resources at the main workshop to carry out the work.

The real workshop would have only one stream and would put coders to work with other attendees in breakouts or hack sessions.  This gathering is not meant to be “restricted", but to be  an important element of commuity participation (perhaps THE central event of project-initiated cadence inclusiveness).  You are proposing travel support, food and alcohol - who could oppose that, let’s take it as a proposal.

Unless I am missing something, the drift is generally or even exactly consistent with your principals and we just need to do a little integration and worsdmithing.

Everybody, am I misreading this?

Steve

**Comment by connolly** [comment]
I think that was the sense of it - I dont think anyone thought it should be coders only but that the pre-workshop might be good for everyone to better understand the technical aspects of what is available for the metric development (and to demonstrate some commitment as we would hope that these people would have sufficient interest to engage in a significant amount of work to get to the whitepaper)

From my perspective I I would like to have something quantitative come out of these sessions. Code is fine, pseudo code is fine, development plans for how to do the experiments to better understand what metrics are needed are also fine (as long as they are detailed and actionable). The hope is that by embedding people with different skills we can get the quantitative aspects as well the science aspects (or at least get the descriptions/plans in an position where they can be coded).

**Comment by bethwillman** [comment]
Hi All -

I agree that we need to have an eye on specific deliverables to come from this workshop.  But I also I think that the vast majority of the community shouldn't have to hear the word MAF.  For a true community connection - People shouldn't need to commit that much effort to meaningfully participate.  

What is the carrot for people to attend?  Interesting and cool stuff.  Energy provided by people with vision being at the same place at the same time.  Maybe a third goal should be made more explicit - to engage more of the general community in this process.

I think our visions might not actually be too different.  One straw man plan, if that third goal is explicitly added:

Day 1 and first morning session of Day 2 - People from the community at large and within the existing collabs, especially those who have objected in the past to any aspect of (what they perceive to be) LSST's plan, give talks that present an issue related to the science cases under scrutiny + related quantifiable cadence needs.  They could be given specific guidance prior to the workshop on how to best present the quantifiable technical needs.  

The evening of Day 1 could be setting up a hack session for the next day - Some pre-selected community members/pundits pitch their ideas, a team of self-identified super users gather around metric(s) they are interested to tackle, and some information is exchanged.  We could engage science collaboration chairs to advertise this opportunity as a way for collaboration members to provide support to the science collaborations.  (People often ask how they can help.)  A combination of sci collab members and Project members could thus form the hack day teams (along the lines of the current vision).

Day 2 - People who will not be hacking metrics attend discussion/talk sessions about (for example) cross facility work, with a couple of hours reserved for checking in with the hack teams to provide clarifying information as needed.

Day 3 morning - A report from the hack sessions and a report from the cross facility conversations.

[Aside 1: How successful are Hack Days in actually delivering completed products?  We should consider how to approach the inevitable fact that work done during the workshop will likely need additional time invested to bring to completion.]

[Aside 2: Steve - I wasn't proposing travel support&food&alcohol, although those are delightful things, just making a comparison with the meetings around other future facilities :) ]

Cheers
Beth

**Comment by djbard** [comment]
@connolly and @bethwillman, what you describe there is very similar to my experience during the 2014 workshop - there was a lot of science discussion which resulted in a list of metrics and pseudo-code (except we never actually got round to the code-writing!). Things may have been different in other groups, of course. If we want to progress beyond what we achieved in the 2014 workshop, I agree that we need hack sessions (even if they don't deliver finished products) but also sessions that produce some language (even if that's only an outline of each section) for the proposed white paper. Beth's proposal is pretty close to that, I think. I'll add it as a starting point over on the [workshop schedule] [1] issue. 
[1]: https://github.com/LSSTScienceCollaborations/ObservingStrategy/issues/4

**Comment by lmwalkowicz** [comment]
To attempt to synthesize the discussion/suggestions here, I think what we are talking about is a two-track workshop where the boundary between the two is a breathable, perforated one: I like the idea of pairing non-super-users with a more technically advanced cadre, both because it makes the workshop more inclusive (thus improving our chances of not excluding anyone with good ideas) and because it expedites producing actual quantitative metrics. Potential attendees can participate in the ways they feel they can be most effective, with the tools of their choice (whether that means the white paper, hacking metrics, or both), and all users have the opportunity to become engaged prior to the workshop itself (and continue after, as well). 

I think we should return to @bethwillman's point though about MAF being the primary way of providing quantitative metrics: that is what I read from this discussion as well (and the takeaway from prior discussions). It seems like @connolly's comment that words are great/pseudocode is great... precisely because specific words and pseudocode can be implemented as metrics in MAF, even if they have not yet. No? I feel like we are having a bit of a disconnect on what constitutes "quantitative feedback" reminiscent of messaging around "feel free to be creative in discussing cadence changes!" that happened in 2014, when really the goal of the workshop was "develop metrics for MAF" not "brainstorm cadence changes that can't currently be evaluated with the available OpSim runs". 

tl;dr: I think quantitative deliverables are great, but I think we should be very clear and specific about what we mean by that so that we can convey it clearly to attendees.

**Comment by StephenRidgway** [comment]
I think it would be great to get out coded metrics, but probably it is unrealistic to require it.  Algorithms contain the intellectual content of the work and is maybe a more realistic minimum metrics contribution.  Why did 2014 not produce this material?  Perhaps because the need was not perceived as imminent.  A reason that I like addressing the white paper development within the workshop is that it constitutes an immediate need for metrics.  The suggestions for hack sessions, breakouts, what have you, all have potential and several options should be available.

**Comment by drphilmarshall** [comment]
See #13 for introducing the above workshop schedule onto the website. I guess we should either get our [notes](https://github.com/LSSTScienceCollaborations/ObservingStrategy/blob/master/workshop/notes/design.md) up to date as well, or remove them from the repo in favor of writing on the website directly.

**Comment by drphilmarshall** [comment]
Rebooting this discussion now that we know from @ivezic in issue #22 that we have [85 people registered for our workshop](https://project.lsst.org/meetings/lsst2015/cadence-registrations).

It looks as though this needn't affect our [proposed schedule](https://github.com/LSSTScienceCollaborations/ObservingStrategy/blob/master/workshop/notes/design.md), but the balance of the parallel sessions may shift.

**Comment by StephenRidgway** [comment]
Here is some take-away information buried in the registration list:

	Of the 85 registered to be in Bremerton on friday, 1/3 are LSST staff or close collaborators
	Of the 51 registered from the general community, only 8 registered for  the 2014 workshop
	Only 4 of 16 “confirmed leads" have registered
	Only 4 of 9 SOC members have registered

	

For what it’s worth, if the total edges up to about the limit of 100, the numbers will be almost exactly the same as August 2014

**Comment by connolly** [comment]
I took a look  through the list and only found 29 from the non-LSST people (and I probably missed the some of the newer people).

**Comment by drphilmarshall** [comment]
There are about 80 people in the room for the first Thursday afternoon session - let's count again after coffee tomorrow (when group work begins! :-) )
